localstack/services/ec2/resource_providers/aws_ec2_natgateway.py

```diff
@@ -0,0 +1,167 @@
+
+        Create-only properties:
+          - /properties/SubnetId
+          - /properties/ConnectivityType
+          - /properties/AllocationId
+          - /properties/PrivateIpAddress
+
+        Read-only properties:
+          - /properties/NatGatewayId
+
+        IAM permissions required:
+          - ec2:CreateNatGateway
+          - ec2:DescribeNatGateways
+          - ec2:CreateTags
+
+        """
+        model = request.desired_state
+        ec2 = request.aws_client_factory.ec2
+
+        # TODO: validations
+        # TODO add tests for this resource at the moment, it's not covered
+
+        if not request.custom_context.get(REPEATED_INVOCATION):
+            # this is the first time this callback is invoked
+            # TODO: defaults
+            # TODO: idempotency
+            params = util.select_attributes(
+                model,
+                ["SubnetId", "AllocationId"],
+            )
+
+            if model.get("Tags"):
+                tags = [{"ResourceType": "natgateway", "Tags": model.get("Tags")}]
+                params["TagSpecifications"] = tags
+
+            response = ec2.create_nat_gateway(
+                SubnetId=model["SubnetId"], AllocationId=model["AllocationId"]
+            )
+            model["NatGatewayId"] = response["NatGateway"]["NatGatewayId"]
+            request.custom_context[REPEATED_INVOCATION] = True
+            return ProgressEvent(
+                status=OperationStatus.IN_PROGRESS,
+                resource_model=model,
+                custom_context=request.custom_context,
+            )
+        response = ec2.describe_nat_gateways(NatGatewayIds=["NAT_GATEWAY_ID"])
+        if response["NatGateways"][0]["State"] == "pending":
+            return ProgressEvent(
+                status=OperationStatus.IN_PROGRESS,
+                resource_model=model,
+                custom_context=request.custom_context,
+            )
+        # TODO add handling for failed events
+        return ProgressEvent(
+            status=OperationStatus.SUCCESS,
+            resource_model=model,
+            custom_context=request.custom_context,
+        )
+
+    def read(
+        self,
+        request: ResourceRequest[EC2NatGatewayProperties],
+    ) -> ProgressEvent[EC2NatGatewayProperties]:
+        """
+        Fetch resource information
+
+        IAM permissions required:
+          - ec2:DescribeNatGateways
+        """
+        raise NotImplementedError
+
+    def delete(
+        self,
+        request: ResourceRequest[EC2NatGatewayProperties],
+    ) -> ProgressEvent[EC2NatGatewayProperties]:
+        """
+        Delete a resource
+
+        IAM permissions required:
+          - ec2:DeleteNatGateway
+          - ec2:DescribeNatGateways
+        """
+        model = request.desired_state
+        ec2 = request.aws_client_factory.ec2
+        """
+        boto3 documentation only has method for deleting single nat gateway
+        instead of the one used bellow
```

Comment:
```suggestion
        instead of the one used below
```

Reply:
Removed comment, as I rewrote function for deleting nat gateway